### DISCUSSION TITLE
Possible to call workflow handler in DSL2 on nested workflow?

### DISCUSSION THREAD

**Discussion by maxbates** [nextflow-io/nextflow#1919] [Q&A]
Is it possible to defined `workflow.onComplete` and `workflow.onError` on a named workflow?

My use-case is a main workflow with a switch statement that calls nested workflows. I'd like to define the handlers on the nested workflows, i.e. the named ones.

Something like:
```
workflow my_nested_workflow {
    ...
}

my_nested_workflow.onComplete {
    ...
}

```

**Comment by Claptar** [comment]
Same here. Seems like `onComplete` and `onError` handlers work only for implicit workflows. It would be great to be able to use them for named ones
